gobblin-test-harness/src/main/java/gobblin/test/setup/config/TestHarnessLauncher.java

```diff
@@ -0,0 +1,57 @@
+package gobblin.test.setup.config;
+
+import gobblin.test.execution.TestHarnessExecutor;
+
+
+/**
+ * An interface for parsing the config data for test harness. 
+ * This interface is allows to parse and validate the config setting
+ * 
+ * @author sveerama
+ *
+ * @param <I> Input folder where the config (JSON/YAML) is stored
+ * @param <C> The reader of the input for the config
+ */
+
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+* this file except in compliance with the License. You may obtain a copy of the
+* License at  http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software distributed
+* under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, either express or implied.
+*/
+public interface TestHarnessLauncher<I, C> extends ConfigStepsGenerator<C>, TestHarnessExecutor<C> {
+
+  /**
+   *  This method will parse the config based on the input provided
+   *
+   * @param inputFolder
+   * @return config
+   *
+   */
+
+  public C parseConfigEntry(I input);
+
+  /**
+   *  This method will validate the config implementer, this allows to customize the config validation
+   *   for each type of config 
+   *
+   * @param <C> config
+   *
+   */
+
+  public void prepareTest(C config);
+
+  /**
+   *  This method will launch the test for config entry after validation
+   *
+   * @param <C> config
+   *
+   */
+
+  public void launchTest(C config);
```

Comment:
So there is a TestHarnessExecutor and a TestHarnessLauncher. Both have a launchTest method? Could you shed some light on the roles of each of these classes?